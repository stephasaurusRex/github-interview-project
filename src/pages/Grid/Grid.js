import get from "lodash.get";
import React, { Component } from "react";
import styled from "styled-components";

import GridLayout from "../../components/GridLayout/GridLayout";
import { fadeColor } from "../../styles";

/* resizable grid layout demo */

class Grid extends Component {
  render() {
    const { className } = this.props;
    return (
      <div className={className}>
        <GridLayout />
      </div>);
  }
}

export default styled(Grid) `
  background: ${(props) => get(props, "theme.blends.grid.accent")};
  color: ${(props) => get(props, "theme.blends.grid.text")};
  min-height: 100%;
  justify-content: center;
  animation: ${(props) =>
    fadeColor(get(props, "theme.blends.grid.accent"), get(props, "theme.blends.grid.primary"))} 1s ease-out forwards;
`;
