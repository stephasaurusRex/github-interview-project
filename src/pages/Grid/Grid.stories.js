import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React, { Fragment } from "react";

import Grid from "./Grid";

storiesOf("grid page", module)
  .add("basic usage", withInfo(`Grid without parameters`)(
    () =>
      <Fragment>
        <Grid></Grid>
      </Fragment>));
