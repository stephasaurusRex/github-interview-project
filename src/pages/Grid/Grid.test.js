import "jest-styled-components";

import React from "react";
import renderer from "react-test-renderer";

import Grid from "./Grid";

describe("Grid", () => {

  it("renders without props", () => {
    const tree = renderer.create(<Grid />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
