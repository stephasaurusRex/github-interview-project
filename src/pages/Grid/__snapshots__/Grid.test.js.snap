// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders without props 1`] = `
.c1 .react-grid-layout {
  position: relative;
  -webkit-transition: height 200ms ease;
  transition: height 200ms ease;
}

.c1 .react-grid-item {
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  -webkit-transition-property: left,top;
  transition-property: left,top;
}

.c1 .react-grid-item.cssTransforms {
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c1 .react-grid-item.resizing {
  z-index: 1;
  will-change: width,height;
}

.c1 .react-grid-item.react-draggable-dragging {
  -webkit-transition: none;
  transition: none;
  will-change: transform;
}

.c1 .react-grid-item.react-grid-placeholder {
  background: mediumpurple;
  opacity: 0.2;
  -webkit-transition-duration: 100ms;
  transition-duration: 100ms;
  z-index: 2;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -o-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c1 .react-grid-item > .react-resizable-handle {
  position: absolute;
  width: 20px;
  height: 20px;
  bottom: 0;
  right: 0;
  cursor: se-resize;
}

.c1 .react-grid-item > .react-resizable-handle::after {
  content: "";
  position: absolute;
  right: 3px;
  bottom: 3px;
  width: 5px;
  height: 5px;
  border-right: 2px solid rgba(0,0,0,0.4);
  border-bottom: 2px solid rgba(0,0,0,0.4);
}

.c0 {
  min-height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-animation: euXcxa 1s ease-out forwards;
  animation: euXcxa 1s ease-out forwards;
}

<div
  className="c0"
>
  <div
    className="c1"
    style={undefined}
  />
</div>
`;
